codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/parse/JsonParserGenerator.kt

```diff
@@ -108,57 +137,14 @@ class JsonParserGenerator(
     override fun operationParser(operationShape: OperationShape): RuntimeType? {
         // Don't generate an operation JSON deserializer if there is no JSON body
         val httpDocumentMembers = httpBindingResolver.responseMembers(operationShape, HttpLocation.DOCUMENT)
-        if (httpDocumentMembers.isEmpty()) {
-            return null
-        }
-
         val outputShape = operationShape.outputShape(model)
         val fnName = symbolProvider.deserializeFunctionName(operationShape)
-        return RuntimeType.forInlineFun(fnName, jsonDeserModule) {
-            it.rustBlockTemplate(
-                "pub fn $fnName(input: &[u8], mut builder: #{Builder}) -> Result<#{Builder}, #{Error}>",
-                "Builder" to outputShape.builderSymbol(symbolProvider),
-                *codegenScope
-            ) {
-                rustTemplate(
-                    """
-                    let mut tokens_owned = #{json_token_iter}(#{or_empty}(input)).peekable();
-                    let tokens = &mut tokens_owned;
-                    #{expect_start_object}(tokens.next())?;
-                    """,
-                    *codegenScope
-                )
-                deserializeStructInner(httpDocumentMembers)
-                expectEndOfTokenStream()
-                rust("Ok(builder)")
-            }
-        }
+        return structureParser(fnName, outputShape, httpDocumentMembers)
     }
 
     override fun errorParser(errorShape: StructureShape): RuntimeType? {
-        if (errorShape.members().isEmpty()) {
-            return null
-        }
         val fnName = symbolProvider.deserializeFunctionName(errorShape) + "json_err"
```

Comment:
Not introduced in this PR, but this should be `+ "_json_err"`.

Reply:
yeah `_json_err` would be slightly nicer but these are private methods—the name only exists to capture the identity of the structure being generated (you could name them a hash of the shape id or something and that would also work fine, the human names are just for debuggability)

Reply:
Fixed in next commit.